octodns/provider/cloudflare.py

```diff
@@ -250,6 +276,12 @@ def _gen_contents(self, record):
         if _type == 'ALIAS':
             _type = 'CNAME'
 
+        # If this is a record to enable to Cloudflare CDN don't update as we
+        # don't know the original values.
+        if (self.cdn and _type == 'CNAME' and
+           record.value.endswith('.cdn.cloudflare.net.')):
+            raise StopIteration
```

Comment:
There's an existing provider mechanism called `_include_change` that's may be more suited to do this sort of omission of specific things.

https://github.com/github/octodns/blob/75403e39f9d721b23fc5cc0ca8243b37543f56a2/octodns/provider/base.py#L29-L34

In fact `CloudflareProvider` actually uses it already to handle the case of their minimum `ttl` being `120s`. 

https://github.com/github/octodns/blob/75403e39f9d721b23fc5cc0ca8243b37543f56a2/octodns/provider/cloudflare.py#L216-L223

You could probably expand the checks there to include the bits you're looking for here and `return False` if there's a match. Both the new case (and probably the existing TTL one) could do with a `self.log.info` to tell people it's happening. Thoughts?